src/GraphQL.AotCompilationSample.TypeFirst/GraphQL.AotCompilationSample.TypeFirst.csproj

```diff
@@ -0,0 +1,33 @@
+<Project Sdk="Microsoft.NET.Sdk">
+
+  <PropertyGroup>
+    <OutputType>Exe</OutputType>
+    <TargetFramework>net7.0</TargetFramework>
+    <ImplicitUsings>enable</ImplicitUsings>
+    <Nullable>enable</Nullable>
+    <PublishAot>true</PublishAot>
```

Comment:
GraphQL.Harness contains <IsPackable>false</IsPackable>. What about here?

Reply:
Which property 'contains false'?

Reply:
FYI, the project fails to AOT-compile if `<IsPackable>false</IsPackable>` exists in the csproj.  I'm not sure how to fix this issue.

```
CVTRES : fatal error CVT1103: cannot read file [C:\Users\Shane\source\repos\graphql-dotnet\graphql-dotnet\src\GraphQL.A
otCompilationSample.TypeFirst\GraphQL.AotCompilationSample.TypeFirst.csproj]
LINK : fatal error LNK1123: failure during conversion to COFF: file invalid or corrupt [C:\Users\Shane\source\repos\gra
phql-dotnet\graphql-dotnet\src\GraphQL.AotCompilationSample.TypeFirst\GraphQL.AotCompilationSample.TypeFirst.csproj]
C:\Program Files\dotnet\sdk\7.0.102\Sdks\Microsoft.DotNet.ILCompiler\build\Microsoft.NETCore.Native.targets(350,5): err
or MSB3073: The command ""C:\Program Files (x86)\Microsoft Visual Studio\2019\Professional\VC\Tools\MSVC\14.29.30133\bi
n\Hostx64\x64\link.exe" @"obj\release\net7.0\win-x64\native\link.rsp"" exited with code 1123. [C:\Users\Shane\source\re
pos\graphql-dotnet\graphql-dotnet\src\GraphQL.AotCompilationSample.TypeFirst\GraphQL.AotCompilationSample.TypeFirst.csp
roj]
```

We certainly don't want nuget packages created for these sample applications.  Perhaps we need a command-line property to in the workflow....?

Reply:
I patched it; see csproj files and workflows.  Let me know if you come up with a better solution.  I did not elect the reverse (special property to get it to build the AOT projects) because otherwise users who clone the repo would not be able to get the AOT samples working.

Reply:
> Which property 'contains false'?

IsPackable. GitHub trimmer did the job 😄

Reply:
> We certainly don't want nuget packages created for these sample applications. Perhaps we need a command-line property to in the workflow....?

DI you check whether a package is really being created on CI? Maybe for such AOT-ed exe it is ignored? I did not check.

Reply:
Yes I checked. Without IsPackable false, it creates a nuget “package” for these projects. (And with it the project fails the AOT build)

Reply:
Something awkward. I checked out this branch, added IsPackable=false settings to both AOT projects and then run dotnet restore/build/pack without issues.

Reply:
```
C:\_GITHUB_\graphql-dotnet\src>dotnet pack --no-restore --no-build -c Release
MSBuild version 17.4.1+9a89d02ff for .NET
  Пакет "C:\_GITHUB_\graphql-dotnet\src\GraphQL.NewtonsoftJson\bin\Release\GraphQL.NewtonsoftJson.7.3.0-preview.nupkg"
  успешно создан.
  Пакет "C:\_GITHUB_\graphql-dotnet\src\GraphQL.DataLoader\bin\Release\GraphQL.DataLoader.7.3.0-preview.nupkg" успешно
  создан.
  Пакет "C:\_GITHUB_\graphql-dotnet\src\GraphQL.MemoryCache\bin\Release\GraphQL.MemoryCache.7.3.0-preview.nupkg" успешн
  о создан.
  Пакет "C:\_GITHUB_\graphql-dotnet\src\GraphQL.MicrosoftDI\bin\Release\GraphQL.MicrosoftDI.7.3.0-preview.nupkg" успешн
  о создан.
  Пакет "C:\_GITHUB_\graphql-dotnet\src\GraphQL.SystemTextJson\bin\Release\GraphQL.SystemTextJson.7.3.0-preview.nupkg"
  успешно создан.
  Пакет "C:\_GITHUB_\graphql-dotnet\src\GraphQL\bin\Release\GraphQL.7.3.0-preview.nupkg" успешно создан.

C:\_GITHUB_\graphql-dotnet\src>
```

Reply:
```
C:\_GITHUB_\graphql-dotnet\src>dotnet --info
ПАКЕТ SDK ДЛЯ .NET:
 Version:   7.0.102
 Commit:    4bbdd14480
```

Reply:
In order to AOT-compile a project, run `dotnet publish`.  `dotnet build` just builds it normally and `dotnet pack` creates nuget packages.

Reply:
From clean project:
```
C:\_GITHUB_\graphql-dotnet\src\GraphQL.AotCompilationSample.TypeFirst>dotnet publish -c Release -o publish --runtime win-x64
MSBuild version 17.4.1+9a89d02ff for .NET
  Определение проектов для восстановления...
  Восстановлен C:\_GITHUB_\graphql-dotnet\src\GraphQL.AotCompilationSample.TypeFirst\GraphQL.AotCompilationSample.Typ
  eFirst.csproj (за 405 ms).
  Проектов обновлено для восстановления: 4 из 5.
  GraphQL -> C:\_GITHUB_\graphql-dotnet\src\GraphQL\bin\Release\net5.0\GraphQL.dll
  GraphQL.MicrosoftDI -> C:\_GITHUB_\graphql-dotnet\src\GraphQL.MicrosoftDI\bin\Release\net5.0\GraphQL.MicrosoftDI.dl
  l
  GraphQL -> C:\_GITHUB_\graphql-dotnet\src\GraphQL\bin\Release\net6.0\GraphQL.dll
  GraphQL.SystemTextJson -> C:\_GITHUB_\graphql-dotnet\src\GraphQL.SystemTextJson\bin\Release\net6.0\GraphQL.SystemTe
  xtJson.dll
  GraphQL -> C:\_GITHUB_\graphql-dotnet\src\GraphQL\bin\Release\netstandard2.0\GraphQL.dll
  GraphQL.StarWars.TypeFirst -> C:\_GITHUB_\graphql-dotnet\src\GraphQL.StarWars.TypeFirst\bin\Release\netstandard2.0\
  GraphQL.StarWars.TypeFirst.dll
  GraphQL.AotCompilationSample.TypeFirst -> C:\_GITHUB_\graphql-dotnet\src\GraphQL.AotCompilationSample.TypeFirst\bin
  \Release\net7.0\win-x64\GraphQL.AotCompilationSample.TypeFirst.dll
  Generating native code
C:\_GITHUB_\graphql-dotnet\src\GraphQL\Attributes\ImplementsAttribute.cs(23): AOT analysis warning IL3050: GraphQL.Im
plementsAttribute.ImplementsAttribute(Type): Using member 'System.Type.MakeGenericType(Type[])' which has 'RequiresDy
namicCodeAttribute' can break functionality when AOT compiling. The native code for this instantiation might not be a
vailable at runtime. [C:\_GITHUB_\graphql-dotnet\src\GraphQL.AotCompilationSample.TypeFirst\GraphQL.AotCompilationSam
ple.TypeFirst.csproj]
<---------------------MANY-MANY WARNINGS HERE ---------------------->
     Creating library bin\Release\net7.0\win-x64\native\GraphQL.AotCompilationSample.TypeFirst.lib and object bin\Rel
  ease\net7.0\win-x64\native\GraphQL.AotCompilationSample.TypeFirst.exp
  GraphQL.AotCompilationSample.TypeFirst -> C:\_GITHUB_\graphql-dotnet\src\GraphQL.AotCompilationSample.TypeFirst\pub
  lish\

C:\_GITHUB_\graphql-dotnet\src\GraphQL.AotCompilationSample.TypeFirst>
```

Reply:
Results:
![изображение](https://user-images.githubusercontent.com/21261007/213291774-c7d38e0d-98fb-4a1c-a8cf-18f2b1a6d93e.png)

GraphQL.AotCompilationSample.TypeFirst.exe works as expected - hero request and introspection request works both.

Reply:
with this in the csproj?

```xml
<Project Sdk="Microsoft.NET.Sdk">

  <PropertyGroup>
    <OutputType>Exe</OutputType>
    <TargetFramework>net7.0</TargetFramework>
    <ImplicitUsings>enable</ImplicitUsings>
    <Nullable>enable</Nullable>
    <PublishAot>true</PublishAot>
    <IsPackable>false</IsPackable>
  </PropertyGroup>
  
  <ItemGroup>
    <PackageReference Include="Microsoft.Extensions.DependencyInjection" Version="7.*" />
    <PackageReference Include="Microsoft.Extensions.Options" Version="7.*" />
  </ItemGroup>

  <ItemGroup>
    <ProjectReference Include="..\GraphQL.MicrosoftDI\GraphQL.MicrosoftDI.csproj" />
    <ProjectReference Include="..\GraphQL.StarWars.TypeFirst\GraphQL.StarWars.TypeFirst.csproj" />
    <ProjectReference Include="..\GraphQL.SystemTextJson\GraphQL.SystemTextJson.csproj" />
  </ItemGroup>

  <ItemGroup>
    <None Remove="IntrospectionQuery.graphql" />
    <EmbeddedResource Include="IntrospectionQuery.graphql" />
  </ItemGroup>
  
  <ItemGroup>
    <TrimmerRootAssembly Include="GraphQL.StarWars.TypeFirst" />
  </ItemGroup>
</Project>

```

Reply:
and this command `dotnet publish -c Release -o out` followed by `out\GraphQL.AotCompilationSample.TypeFirst.exe`?

Reply:
![image](https://user-images.githubusercontent.com/6377684/213293164-d10f401e-ab5b-4f92-ad88-49dc2d1d4e89.png)

Reply:
I am using .NET 7.0.102

Reply:
Yes. I did it again, copied aforementioned csproj xml, cleaned obj/bin/out, then `dotnet publish -c Release -o out` followed by `out\GraphQL.AotCompilationSample.TypeFirst.exe`. It works.

Reply:
![изображение](https://user-images.githubusercontent.com/21261007/213296753-2cbd0b15-c06f-4a46-941e-587216b9d1f5.png)


from https://gitter.im/dotnet/corert?at=5de7a3431659720ca8f8a2b2

Reply:
https://github.com/dotnet/runtimelab/issues/1436

Reply:
I reviewed PR 4-5 times. This is only conversation left.

Reply:
Looks like deleting the obj folder fixed the issue. 👍

Reply:
Removed condition, removed changes to workflow, added `<IsPackable>false</IsPackable>`